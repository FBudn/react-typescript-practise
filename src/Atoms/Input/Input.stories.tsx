import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Input from "./Input";

export default {
  title: "Atoms/Input",
  component: Input,
  argTypes: {
    ref: { action: "I'm forwarding a reference" },
  },
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;

export const FlexFullWidth = Template.bind({});
FlexFullWidth.args = {
  type: "text",
};
