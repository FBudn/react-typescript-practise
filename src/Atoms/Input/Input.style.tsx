import styled from "styled-components";

export const InputContainer = styled.input`
  display: flex;
  width: 100%;
  border-radius: 5px;
  height: 40px;
  border: 2px solid rgb(214 211 209);
`;
