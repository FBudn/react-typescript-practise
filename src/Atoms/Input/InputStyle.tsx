import styled from "styled-components";

export const InputAtom = styled.input`
  display: flex;
  border-radius: 5px;
  height: 40px;
  border: 2px solid rgb(214 211 209);
`;
