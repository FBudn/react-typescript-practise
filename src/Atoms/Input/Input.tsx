import React, { forwardRef } from "react";
import { InputContainer } from "./Input.style";

export interface InputProps {
  type: string;
  testId?: string;
}

export type InputRef = HTMLInputElement;
const Input = forwardRef<InputRef, InputProps>(({ type, testId }, ref) => (
  <InputContainer data-testid={testId} type={type} ref={ref} />
));

export default Input;
