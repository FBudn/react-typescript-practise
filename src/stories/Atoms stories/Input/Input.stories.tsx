import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Input from "../../../Atoms/Input/Input";

export default {
  title: "Atoms/Input",
  component: Input,
  argTypes: {
    ref: { action: "I'm forwarding a reference" },
  },
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;

export const FlexFullWidth = Template.bind({});
FlexFullWidth.args = {
  type: "text",
};

// roznica w componentMeta vs as Meta po prostu i tak samo w Story
// szerokość inputa co tutaj
