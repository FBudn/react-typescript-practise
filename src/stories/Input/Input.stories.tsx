/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/jsx-props-no-spreading */
import React, { useRef } from "react";
import Input from "../../Atoms/Input/Input";

export default {
  title: "Atoms/Input",
  component: Input,
  argTypes: {
    useRef: useRef,
    type: "text",
  },
};

const Template = (args) => <Input {...args} />;

export const TextInputBasic = Template.bind({});
TextInputBasic.args = {
  onClick: "jestem",
};
