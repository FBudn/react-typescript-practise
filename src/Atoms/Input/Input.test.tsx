/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
import React from "react";
import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Input, { InputProps } from "./Input";
import "jest-styled-components";

const props: InputProps = {
  type: `text`,
  testId: `test-input-id`,
};

test(`Should render Input component`, () => {
  render(<Input {...props} />);
  const InputElement = screen.getByTestId(`test-input-id`);
  expect(InputElement).toBeInTheDocument();
});

test(`Should match snapshot`, () => {
  const InputElement = TestRenderer.create(<Input {...props} />).toJSON();
  expect(InputElement).toMatchSnapshot();
});

test(`Should render and have styles:`, () => {
  render(<Input {...props} />);
  const InputElement = screen.getByTestId(`test-input-id`);
  expect(InputElement).toBeInTheDocument();

  expect(InputElement).toHaveStyle({
    display: "flex",
    width: "99%",
    borderRadius: "5px",
    height: "40px",
  });
});

test(`Should store input value`, () => {
  render(<Input {...props} />);
  const InputElement = screen.getByTestId(`test-input-id`) as HTMLInputElement;
  InputElement.value = "Test";
  expect(InputElement).toHaveValue("Test");
});
