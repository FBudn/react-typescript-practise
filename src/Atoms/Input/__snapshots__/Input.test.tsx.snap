// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 99%;
  border-radius: 5px;
  height: 40px;
  border: 2px solid rgb(214 211 209);
}

<input
  className="c0"
  data-testid="test-input-id"
  type="text"
/>
`;
